nautobot/extras/api/customfields.py

```diff
@@ -50,10 +58,34 @@ def _get_custom_fields(self):
         return self._custom_fields
 
     def to_representation(self, obj):
-        return {cf.name: obj.get(cf.name) for cf in self._get_custom_fields()}
+        request = self.context.get("request")
+        if request and (request.major_version > 1 or request.minor_version >= 4):
+            # 1.4+ behavior
+            # 2.0 TODO: #824 use cf.slug as lookup key instead of cf.name
+            return {cf.slug: obj.get(cf.name) for cf in self._get_custom_fields()}
+        else:
+            # Legacy behavior
+            return {cf.name: obj.get(cf.name) for cf in self._get_custom_fields()}
 
     def to_internal_value(self, data):
         """Support updates to individual fields on an existing instance without needing to provide the entire dict."""
+        request = self.context.get("request")
+        if request and (request.major_version > 1 or request.minor_version >= 4):
+            # 1.4+ behavior
+            use_slug = True
+        else:
+            # Legacy behavior
+            use_slug = False
+
+        if use_slug:
+            # Map slugs to names for the backend data
+            # 2.0 TODO: #824 remove this translation
+            new_data = {}
+            for slug, value in data.items():
+                cf = CustomField.objects.get(slug=slug)
+                new_data[cf.name] = value
```

Comment:
Maybe a slight optimization to get all the `CustomFields` in a  single `filter` query and peel them off in the iterator so there aren't a bunch of single row queries?

```suggestion
            new_data = {}
            custom_fields = CustomField.objects.filter(slug__in=data.keys())
            for cf in custom_fields.iterator():
                new_data[cf.name] = data[cf.slug]
```